app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -78,72 +76,51 @@ class TransferSendManager @Inject constructor(
     private suspend fun sendTransfer(uploadSessionUuid: String) {
         _sendStatus.value = SendStatus.Pending
 
-        withIntegrityToken(
-            onSuccess = { attestationToken ->
-                runCatching {
-                    uploadManager.initUploadSession(
-                        attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
-                        attestationToken = attestationToken,
-                    )!! // TODO: Handle ContainerErrorsException here
-                    uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
-                    _sendStatus.update {
-                        val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
-                        SendStatus.Success(totalSize)
-                    }
-                }.onFailure { exception ->
-                    SentryLog.e(TAG, "Failed to start the upload", exception)
-                    _sendStatus.update { SendStatus.Failure }
-                }
-            },
-            onRefused = { _sendStatus.update { SendStatus.Refused } },
-        )
+        runCatching {
+            val attestationToken = getAttestationToken()
+            uploadManager.initUploadSession(
+                attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
+                attestationToken = attestationToken,
+            )!! // TODO: Handle ContainerErrorsException here
+            uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
+            _sendStatus.update {
+                val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
+                SendStatus.Success(totalSize)
+            }
+        }.onFailure { exception ->
+            if (exception !is NetworkException && exception !is KmpNetworkException) {
```

Comment:
Maybe we should add a different result for network exception to prompt the user that he has no connection instead of just "an error has occured" where it seems the problem is on our side